src/Illuminate/Database/Schema/Grammars/SqlServerGrammar.php

```diff
@@ -105,10 +105,11 @@ public function compilePrimary(Blueprint $blueprint, Fluent $command)
      */
     public function compileUnique(Blueprint $blueprint, Fluent $command)
     {
-        return sprintf('create unique index %s on %s (%s)',
+        return sprintf('create unique index %s on %s (%s) where "%s" is not null',
```

Comment:
Minor nitpick: The quotes should come from `wrap()`.

```suggestion
        return sprintf('create unique index %s on %s (%s) where %s is not null',
```